naima/models.py

```diff
@@ -431,3 +431,87 @@ def __call__(self, e):
         e = _validate_ene(e)
         interpy = np.power(10, self._interplogy(np.log10(e.to('eV').value)))
         return self.amplitude * interpy * self.unit
+      
+      
+class AbsorptionModel(object):
+    """
+    A Table-model like class containing the different absorption models.
+
+    It returns dimensionless opacity values, that could be multiplied to any model.
+    
+    Parameters
+    ----------    
+    redshift : float
+        Redshift considered for the absorption evaluation.
+        The minimum value used is z = 0.01; Lower values will
+        be considered as redshift 0.
+        
+    ebl_absorption_model : string 
+        Name of the EBL absorption model to use. It may be:
+    
+    * A string equal to ``Dominguez`` (default), containing Dominguez 2011 EBL model
+    
+    NOTE: Current implementation does NOT perform an interpolation in the redshift, so 
+    it just uses the closes z value from the finely binned tau_dominguez11.out file 
+    (delta_z=0.01)    
+        
+    """
+
+    def __init__(self, redshift, ebl_absorption_model='Dominguez'):
+        from scipy.interpolate import interp1d
+        from astropy.io import ascii
+                    
+        if ebl_absorption_model == 'Dominguez':
+            converters = dict()
+            for i in range(0,500):
+               converters["col%s" % i] = [ascii.convert_numpy(np.float64)]
+            taus_table = ascii.read('naima/data/tau_dominguez11.out', converters=converters)
+            redshift_list = np.arange(0.01,4,0.01)
+            taus_table['col1'].name = 'energy'      
+            taus_table['energy'].unit = u.TeV      
+            energy = taus_table['energy'].quantity
+            if (redshift >= 0.01):
+                taus_table['col%s' % (2+(np.abs(redshift_list-redshift)).argmin())].name = 'tau'
+                taus_table['tau'].unit = u.dimensionless_unscaled
+                taus = taus_table['tau'].quantity
```

Comment:
is there a reason you can't directly do `taus = taus_table['col%s' % (2+(np.abs(redshift_list-redshift)).argmin())].quantity`?